aya/src/programs/common/btf_id.rs

```diff
@@ -0,0 +1,40 @@
+//! Common functions for all eBPF program types that attach to a BTF id.
+use std::{
+    ffi::{CStr, CString},
+    os::unix::io::RawFd,
+};
+
+use crate::{
+    programs::{FdLink, LinkRef, ProgramData, ProgramError},
+    sys::bpf_raw_tracepoint_open,
+};
+
+fn attach_btf_id_cstr(
```

Comment:
A couple of things.

1. I would probably put this in `aya/src/programs/utils.rs` rather than creating a new subdir.
1. I think this fn should be `attach_btf_id` and then it's up to the caller to convert their `Option<&str>` to `Option<&cstr>`

Reply:
Good idea, done